synfig-studio/src/gui/duckmatic.cpp

```diff
@@ -856,35 +847,60 @@ Duckmatic::snap_point_to_grid(const synfig::Point& x)const
 			floor(ret[1]/get_grid_size()[1]+0.5)*get_grid_size()[1]);
-
-		if(std::fabs(snap[0]-ret[0])<=radius && (!has_guide_x || std::fabs(snap[0]-ret[0])<=std::fabs(*guide_x-ret[0])))
-			ret[0]=snap[0],has_guide_x=false;
-		if(std::fabs(snap[1]-ret[1])<=radius && (!has_guide_y || std::fabs(snap[1]-ret[1])<=std::fabs(*guide_y-ret[1])))
-			ret[1]=snap[1],has_guide_y=false;
-	}
-
-	if(guide_snap)
-	{
-		if(has_guide_x)
-			ret[0]=*guide_x;
-		if(has_guide_y)
-			ret[1]=*guide_y;
+		float distance_from_guide = calculate_distance_from_guide(*guide, ret);
+		if(std::fabs(snap[0]-ret[0])<=radius && (!has_guide || std::fabs(snap[0]-ret[0])<=distance_from_guide))
+			ret[0]=snap[0],has_guide=false;
+		if(std::fabs(snap[1]-ret[1])<=radius && (!has_guide || std::fabs(snap[1]-ret[1])<=distance_from_guide))
+			ret[1]=snap[1],has_guide=false;
+	}
+
+	if(guide_snap && has_guide)
+	{
+		bool possible_intersection_close = second_best_guide_match != nullptr && (second_best_guide_match->angle != guide->angle);
+		if (possible_intersection_close){
+					float guides_intersection_x = 0, guides_intersection_y = 0;
+					if (guide->angle != synfig::Angle::deg(90) && second_best_guide_match->angle != synfig::Angle::deg(90)){
+						float slope_best = tan(guide->angle.get()), slope_second_best_guide_match = tan(second_best_guide_match->angle.get());
+						float bestX = guide->point[0], bestY = guide->point[1], lastBestX = second_best_guide_match->point[0], lastBestY = second_best_guide_match->point[1];
+						guides_intersection_x = ( bestY - lastBestY + slope_second_best_guide_match*lastBestX - slope_best*bestX )/(slope_second_best_guide_match - slope_best);
+						guides_intersection_y = slope_best*(guides_intersection_x - bestX) + bestY;
+						possible_intersection_close = (pow(pow(ret[1] - guides_intersection_y, 2.0) + pow(ret[0] - guides_intersection_x, 2.0), 0.5)) <= radius*3;
+					} else if (guide->angle != second_best_guide_match->angle){
+						if (synfig::Angle::deg(guide->angle).get() == 90){
+							guides_intersection_x = guide->point[0];
+							guides_intersection_y = tan(second_best_guide_match->angle.get())*(second_best_guide_match->point[0]) + second_best_guide_match->point[1];
+						} else if (synfig::Angle::deg(second_best_guide_match->angle).get() == 90){
+							guides_intersection_x = second_best_guide_match->point[0];
+							guides_intersection_y = tan(guide->angle.get())*(guide->point[0]) + guide->point[1];
+						}
+						possible_intersection_close = (pow(pow(ret[1] - guides_intersection_y, 2.0) + pow(ret[0] - guides_intersection_x, 2.0), 0.5)) <= radius*3;
+					}
+					if (possible_intersection_close) {
+						ret[0] = guides_intersection_x;
+						ret[1] = guides_intersection_y;
+					}
+				}
+		if (!possible_intersection_close && guide->angle == synfig::Angle::deg(90))
+			ret[0]=guide->point[0];
+		else if (!possible_intersection_close && guide->angle == synfig::Angle::deg(0))
+			ret[1]=guide->point[1];
```

Comment:
surround assignment operator with space ` = `

Reply:
the previous code in this method has this style without spaces should I edit that as well ? or maybe refactor the other code in another pr ? because I think having the same method have different styles would probably not be a good thing.

Reply:
Alright then, done xD.